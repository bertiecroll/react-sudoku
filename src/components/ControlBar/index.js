export { ControlBar } from './ControlBar'
export { default } from './ControlBar'
