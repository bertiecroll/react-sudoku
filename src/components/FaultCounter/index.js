export { FaultCounter } from './FaultCounter'
export { default } from './FaultCounter'
