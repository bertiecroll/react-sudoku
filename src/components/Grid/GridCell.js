import React from 'react'
import PropTypes from 'prop-types'
import './Grid.css'

export const GridCell = (props) => {
  return (
    <div className="GridCell">
      { props.number }
    </div>
  )
}

GridCell.propTypes = {
  number: PropTypes.number.isRequired
}
