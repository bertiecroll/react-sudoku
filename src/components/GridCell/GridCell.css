.GridCell {
  width: 2.5em;
  height: 2.5em;
  padding: .125em;
  text-align: center;
}

.GridCell:hover {
  cursor: pointer;
}

.GridCell:first-child {
  border-left: none;
}

.GridCell:nth-child(3n) {
  padding-right: .125em;
  border-right: 5px solid #777;
}

.GridCell:nth-child(3n + 1) {
  padding-left: .25em;
}

.GridCell:last-child {
  border-right: none;
}

.GridCell div {
  width: 2.4em;
  height: 2.4em;
  border-radius: 5px;
  line-height: 2.4em;
}

.empty {
  display: flex;
  flex-wrap: wrap;
  background-color: #ccc;
  box-shadow: inset 0 0 2px #000;
}

.empty .pencil-marks {
  width: 1.25em;
  height: 1.25em;
  line-height: 1.25em;
  font-size: 10px;
  color: white;
}

.complete {
  background-color: #67DAF9;
  box-shadow: 0 1px 1px 0 #000;
  color: #fff;
}

.cell-selected {
  box-shadow: 0 0 2px 2px darkblue;
}

.cell-selected .fault {
  color: darkblue;
  transform: scale(5);
  padding-top: 0.05em;
}
