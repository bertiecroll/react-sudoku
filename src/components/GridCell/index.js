export { GridCell } from './GridCell'
export { EmptyGridCell } from './EmptyGridCell'
export { CompleteGridCell } from './CompleteGridCell'
export { default } from './GridCell'
