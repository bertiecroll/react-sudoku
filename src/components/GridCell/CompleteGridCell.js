import React from 'react'
import PropTypes from 'prop-types'

import GridCell from 'components/GridCell'

export function CompleteGridCell ({ cell }) {
  return <GridCell key={cell.id} cell={cell} render={_renderCompleteGridCell} />
}

function _renderCompleteGridCell ({cell, selectedCell, onCellClick }) {
  const { completed, value } = selectedCell
  const cellClassNames = `complete ${(completed && cell.value === value) ? 'cell-selected' : ''}`

  return <div className={cellClassNames} onClick={onCellClick} >{ cell.value }</div>
}

CompleteGridCell.defaultProps = { cell: {} }
CompleteGridCell.propTypes = { cell: PropTypes.object.isRequired }
