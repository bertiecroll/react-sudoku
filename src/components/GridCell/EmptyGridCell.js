import React from 'react'
import PropTypes from 'prop-types'
import map from 'lodash/fp/map'
import includes from 'lodash/fp/includes'

import GridCell from 'components/GridCell'

export function EmptyGridCell ({ cell }) {
  return <GridCell cell={cell} render={_renderEmptyGridCell} />
}

function _renderEmptyGridCell ({ cell, selectedCell, faultCellId, onCellClick }) {
  const hasFault = faultCellId === cell.id
  const pencilMarkExists = includes(selectedCell.value)(cell.pencilMarks)
  const isSelected = (cell === selectedCell || (selectedCell.completed && pencilMarkExists))
  const cellClassNames = isSelected ? 'empty cell-selected' : 'empty'

  return (
    <div className={cellClassNames} onClick={onCellClick} >
      {hasFault && <div className="fault">X</div>}
      {map((number, key) => <div className="pencil-marks" key={key}>{number}</div>)(cell.pencilMarks)}
    </div>
  )
}

EmptyGridCell.defaultProps = { cell: {} }
EmptyGridCell.propTypes = { cell: PropTypes.object.isRequired }
