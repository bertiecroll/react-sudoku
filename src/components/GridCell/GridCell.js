import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import './GridCell.css'
import { updateSelectedCellId } from 'redux/actions/grid'
import { selectedCell } from 'redux/selectors/cells'
import { faultCellId } from 'redux/selectors/fault'

export function GridCell ({ render, cell, selectedCell, faultCellId, updateSelectedCellId }) {
  return (
    <div className="GridCell">
      { render({ cell, selectedCell, faultCellId, onCellClick: () => updateSelectedCellId(cell.id) }) }
    </div>
  )
}

const mapStateToProps = state => ({ selectedCell: selectedCell(state), faultCellId: faultCellId(state) })
const mapDispatchToProps = { updateSelectedCellId }

export default connect(mapStateToProps, mapDispatchToProps)(GridCell)

GridCell.defaultProps = {
  selectedCell: {}
}

GridCell.propTypes = {
  faultCellId: PropTypes.string,
  selectedCell: PropTypes.object.isRequired,
  updateSelectedCellId: PropTypes.func.isRequired,
}
