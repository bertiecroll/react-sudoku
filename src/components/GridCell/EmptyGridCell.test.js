import React from 'react'
import { shallow } from 'enzyme'
import { EmptyGridCell } from 'components/GridCell'

describe('EmptyGridCell', () => {
  it('renders GridCell passing render prop function', () => {
    const cell = { id: '0,0', value: 1, pencilMarks: [1, 2, 3] }
    const wrapper = shallow(<EmptyGridCell cell={cell} />)

    expect(wrapper).toMatchSnapshot()
  })
})
