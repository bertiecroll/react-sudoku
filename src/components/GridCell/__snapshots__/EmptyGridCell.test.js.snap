// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyGridCell renders GridCell passing render prop function 1`] = `
<Connect(GridCell)
  cell={
    Object {
      "id": "0,0",
      "pencilMarks": Array [
        1,
        2,
        3,
      ],
      "value": 1,
    }
  }
  render={[Function]}
/>
`;
